### PR TITLE
fix[vortex-array]: handle lazily sliced REE to_arrow

### PR DESCRIPTION
**Pull request by asubiotto**
## Summary

This previous code was short-circuiting with a zero offset, with the assumption that ends were already correctly physically sliced. However, slicing is a lazy operation using metadata, so this commit fixes a couple of edge cases by executing the fast path only if the ends are correcly sliced. Otherwise, ends are trimmed and transformed with the offset in for correct arrow array construction.

## Testing

Enhanced tests to reproduce e2e arrow execution error and added some unit tests to build_run_array

### CODE REVIEW

File: vortex-array/src/arrow/executor/run_end.rs [hunk 1]
```diff
@@ -118,22 +118,35 @@ fn build_run_array<R: RunEndIndexType>(
...
+
+    // Trim to only include runs covering the [offset, offset+length) range.
+    // Runs beyond this would produce duplicate adjusted ends, violating
+    // Arrow's strict-ordering requirement for RunArray.
+    let num_runs = ends_prim
+        .values()
+        .iter()
+        .position(|&e| e - offset_native >= length_native)
+        .map(|i| i + 1)
+        .unwrap_or(ends_prim.len());
```
Comment (joseph-isaacs) [inline vortex-array/src/arrow/executor/run_end.rs]: can we binary search for this?

Comment (joseph-isaacs) [reply vortex-array/src/arrow/executor/run_end.rs]: maybe `partition_point`?

Comment (asubiotto) [reply vortex-array/src/arrow/executor/run_end.rs]: Done, thanks

**Comment by asubiotto** [comment]
Updated to binary search.

**Review by joseph-isaacs — APPROVED** [review]
LG, thanks
